clients/python/girder_client/cli.py

```diff
@@ -39,14 +40,27 @@ def __init__(self, username, password, host=None, port=None, apiRoot=None,
         :param password: password to authenticate to Girder instance, leave
             this blank to be prompted.
         """
+        if not any((apiUrl, apiRoot, host, port, scheme)) and config:
+            apiUrl = config.get('girder_client', 'apiUrl')
         super(GirderCli, self).__init__(
             host=host, port=port, apiRoot=apiRoot, scheme=scheme, apiUrl=apiUrl)
-        interactive = password is None
+        interactive = password is None and \
+            (config is None or config.get("girder_client", "password") is None)
 
         if apiKey:
             self.authenticate(apiKey=apiKey)
         elif username:
             self.authenticate(username, password, interactive=interactive)
+        elif config:
+            configApiKey = config.get('girder_client', 'apiKey')
+            if configApiKey:
+                self.authenticate(apiKey=configApiKey)
+            elif config.get('girder_client', 'username'):
+                self.authenticate(config.get('girder_client', 'username'),
+                                  config.get('girder_client', 'password'),
+                                  interactive=False)
+        else:
+            print('Unable to authenticate. Insufficient options')
```

Comment:
No, actually the `else` statement should be removed. You can still download public folder if you are not authenticated user.